### PR TITLE
Unify logging

### PR DESCRIPTION
**Pull request by TonyB9000**

### CODE REVIEW

File: warehouse/warehouse/scripts/check_time_units.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def main():
                 with open(messages_path, 'w') as outstream:
                     outstream.write(message.replace(':', '^'))
             else:
-                con_message(LEVEL,message)
+                con_message('error',message)  # no idea if this should be info, warning or error
```
Comment (sterlingbaldwin) [inline warehouse/warehouse/scripts/check_time_units.py:94]: space

File: warehouse/warehouse/scripts/fix_time_units.py [hunk 2]
```diff
@@ -11,8 +11,9 @@ def fix_units(inpath, outpath, time_units, offset):
     import xarray as xr
     with xr.open_dataset(inpath, decode_times=False) as ds:
         if ds.get('time') is None:
-            con_message('error',f"{inpath} has no 'time' axis")
-            raise ValueError(f"{inpath} has no 'time' axis")
+            con_message('error',f"{os.path.basename(inpath)} has no 'time' axis")
+            exit(1)
+            # raise ValueError(f"{inpath} has no 'time' axis")
```
Comment (sterlingbaldwin) [inline warehouse/warehouse/scripts/fix_time_units.py:16]: remove the comment, the previous version is recorded in git if we need to reference it

File: warehouse/warehouse/workflows/publication/__init__.py [hunk 3]
```diff
@@ -73,6 +73,7 @@ def __call__(self, *args, **kwargs):
         warehouse.start_listener()
 
         for dataset_id, dataset in warehouse.datasets.items():
+            log_message('info',f'starting job {self.name} for {dataset_id}')
```
Comment (sterlingbaldwin) [inline warehouse/warehouse/workflows/publication/__init__.py:76]: this should be "starting workflow"

**Review by sterlingbaldwin — CHANGES_REQUESTED** [review]
looks good, except two things

- spaces after the comma in multi-argument function calls
- remove all the commented code

File: warehouse/warehouse/scripts/check_time_units.py (hunk 1 above)
Comment (TonyB9000) [reply warehouse/warehouse/scripts/check_time_units.py:94]: I'll do a thorough blacking this morning.

File: warehouse/warehouse/scripts/fix_time_units.py (hunk 2 above)
Comment (TonyB9000) [reply warehouse/warehouse/scripts/fix_time_units.py:16]: ok, sure.

File: warehouse/warehouse/workflows/publication/__init__.py (hunk 3 above)
Comment (TonyB9000) [reply warehouse/warehouse/workflows/publication/__init__.py:76]: Will fix.

Comment (TonyB9000) [reply warehouse/warehouse/workflows/publication/__init__.py]: OK, I think I've got it all - or mostly/sort-of...  Please review!

File: esgfpub/scripts/bart_code/warehouse_status.py [hunk 4]
```diff
@@ -276,9 +276,12 @@ def get_ensemble_dirs():
 # into dictionary, key = STAT, rows are tuples (ts,'PROCESS:status1:status2:...')
 # and for comments, key = COMM, rows are comment lines
 
+gv_status_root = "/p/user_pub/e3sm/staging/status"
+
 def load_DatasetStatusFile(edir):
     statdict = {}
-    statfile = os.path.join(edir,'.status')
+    dsid = get_dsid(edir)
+    statfile = os.path.join(gv_status_root,dsid + '.status')
```
Comment (sterlingbaldwin) [inline esgfpub/scripts/bart_code/warehouse_status.py:284]: ...

File: warehouse/warehouse/scripts/move_to_publication.py [hunk 5]
```diff
@@ -77,7 +80,7 @@ def conduct_move(args):
         with open(messages_path, 'w') as outstream:
             outstream.write(message)
     else:
-        con_message('info',message)
+        con_message('error',message)
```
Comment (sterlingbaldwin) [inline warehouse/warehouse/scripts/move_to_publication.py:83]: ...

File: warehouse/warehouse/scripts/rectify_time_index.py [hunk 6]
```diff
@@ -63,85 +68,97 @@ def monotonic_check(path, idx, bndsname):
...
+                )
                 return None, None, idx
 
         return start_bound, end_bound, idx
 
 
 def collect_segments(inpath, num_jobs, timename, bndsname):
 
-    con_message('info',"starting segment collection")
+    con_message("info", "starting segment collection")
```
Comment (sterlingbaldwin) [inline warehouse/warehouse/scripts/rectify_time_index.py:82]: yay!

File: warehouse/warehouse/scripts/validate_esgf.py [hunk 7]
```diff
@@ -20,12 +20,11 @@ def parse_args():
     return parser.parse_args()
 
 
-def publish_dataset(dataset_id: str):
+def publish_dataset(dataset_id: str):   # should be renamed
```
Comment (sterlingbaldwin) [inline warehouse/warehouse/scripts/validate_esgf.py:23]: Its OK to rename this

File: warehouse/warehouse/workflows/__init__.py [hunk 8]
```diff
@@ -106,8 +107,9 @@ def next_state(self, dataset, state, params, idx=1):
             return self.children[state_attrs[idx]].next_state(dataset, state, params, idx + 1)
 
         else:
-            raise ValueError(
-                f"{target_state} is not present in the transition graph for {self.name}")
+            log_message('error', f"{target_state} is not present in the transition graph for {self.name}")
+            sys.exit(1)
+            # raise ValueError(f"{target_state} is not present in the transition graph for {self.name}")
```
Comment (sterlingbaldwin) [inline warehouse/warehouse/workflows/__init__.py:112]: remove commented-out code.

**Review by sterlingbaldwin — CHANGES_REQUESTED** [review]

File: warehouse/warehouse/workflows/__init__.py (hunk 8 above)
Comment (TonyB9000) [reply warehouse/warehouse/workflows/__init__.py]: done

File: warehouse/warehouse/workflows/__init__.py [hunk 9]
```diff
@@ -165,8 +167,9 @@ def load_children(self):
 
             module_path = Path(my_path, d.name, '__init__.py')
             if not module_path.exists():
-                raise ValueError(
-                    f"{module_path} doesnt exist, doesnt look like this is a well formatted workflow")
+                log_message('error', f"{module_path} doesnt exist, doesnt look like this is a well formatted workflow")
+                sys.exit(1)
+                # raise ValueError( f"{module_path} doesnt exist, doesnt look like this is a well formatted workflow")
```
[the remark this answers is not in the document]
Comment (TonyB9000) [reply warehouse/warehouse/workflows/__init__.py]: done

File: warehouse/warehouse/scripts/validate_esgf.py (hunk 7 above)
Comment (TonyB9000) [reply warehouse/warehouse/scripts/validate_esgf.py]: done

File: warehouse/warehouse/scripts/rectify_time_index.py (hunk 6 above)
Comment (TonyB9000) [reply warehouse/warehouse/scripts/rectify_time_index.py:84]: done (forgot to push)

File: warehouse/warehouse/scripts/move_to_publication.py (hunk 5 above)
Comment (TonyB9000) [reply warehouse/warehouse/scripts/move_to_publication.py]: done (forgot to push)

File: warehouse/warehouse/scripts/move_to_publication.py [hunk 10]
```diff
@@ -57,13 +57,16 @@ def conduct_move(args):
     src_path = Path(args.src)
     dst_path = Path(args.dst)
 
+    fc = 0
```
[the remark this answers is not in the document]
Comment (TonyB9000) [reply warehouse/warehouse/scripts/move_to_publication.py]: done

File: esgfpub/scripts/bart_code/warehouse_status.py (hunk 4 above)
Comment (TonyB9000) [reply esgfpub/scripts/bart_code/warehouse_status.py:284]: done (forgot to push)

**Comment by TonyB9000** [comment]
I think I've got it all - am I supposed to push some other button now?

**Comment by sterlingbaldwin** [comment]
I cleaned up the last issues with missing spaces, and cleaned up the code a bit as I was looking through. Added some comments to methods and added some log messages.

**Comment by TonyB9000** [comment]
Great.  Sorry about the spaces (black black black . . .)...

**Pull request merged**